### PR TITLE
Query: Fixes bug in OrderByCrossPartitionQueryPipelineStage to ensure that errors in inner pipeline creation are bubbled up

### PR DESCRIPTION
**Pull request by neildsh**
## Description

Fixes bug in OrderByCrossPartitionQueryPipelineStage to ensure that errors in inner pipeline creation are bubbled up to the higher stages

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/OrderByQueryTests.cs [hunk 1]
```diff
@@ -1181,6 +1181,58 @@ public async Task TestMixedTypeOrderByAsync()
...
+                foreach (bool enableOptmisticDirectExecution in new []{ false, true })
+                {
+                    QueryRequestOptions options = new QueryRequestOptions()
+                    {
+                        EnableOptimisticDirectExecution = enableOptmisticDirectExecution,
+                    };
+
+                    foreach ((string query, string error) in queries)
+                    {
+                        using FeedIterator feedIterator = container.GetItemQueryStreamIterator(query, continuationToken: null, options);
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/OrderByQueryTests.cs:1222]: >GetItemQueryStreamIterator [](http://example.com/codeflow?start=68&length=26)

I am not 100% sure whether ODE kicks in for Stream API. We had a pending item for this. Although pipeline construction seems agnostic of top level api so may be that item is a no-ip.

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs [hunk 2]
```diff
@@ -160,7 +160,7 @@ public async ValueTask<bool> MoveNextAsync(ITrace trace, CancellationToken cance
             }
 
             TryCatch<bool> hasNext = await this.inner.TryAsync(pipelineStage => pipelineStage.MoveNextAsync(trace, cancellationToken));
-            return hasNext.Succeeded && hasNext.Result;
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs:163]: I stared at this line the other day for good 10 minutes and didn't find anything wrong. I find the TryCatch implementation coupled with pipeline's MoveNextAsync return value contract non-intuitive.

**Review by adityasa — APPROVED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/OrderByQueryTests.cs (hunk 1 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/OrderByQueryTests.cs:1222]: Maybe this is the reason why an error is being produced in ODE case as well.

Comment (neildsh) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Query/OrderByQueryTests.cs:1222]: I verified that the ODE pipeline is kicking in for these test cases

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs (hunk 2 above)
Comment (neildsh) [reply Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs:163]: Agreed. This is really easy to miss.

**Review by Maya-Painter — APPROVED** [review]

**Pull request merged**